src/filesystem/impls/filer/FileSystemCache.js

```diff
@@ -0,0 +1,78 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Content = require("filesystem/impls/filer/lib/content");
+    var async = require("filesystem/impls/filer/lib/async");
+    var BracketsFiler = require("filesystem/impls/filer/BracketsFiler");
+    var BlobUtils = require("filesystem/impls/filer/BlobUtils");
+    var Path = BracketsFiler.Path;
+    var Transforms = require("filesystem/impls/filer/lib/transforms");
+    var StartupState = require("bramble/StartupState");
+    var decodePath = require("filesystem/impls/filer/FilerUtils").decodePath;
+
+    // Walk the project root dir and make sure we have Blob URLs generated for
+    // all file paths.
+    exports.refresh = function(callback) {
+        var fs = BracketsFiler.fs();
+
+        function _getUrlAsync(filename, callback) {
+            var decodedFilename = decodePath(filename);
+            var cachedUrl = BlobUtils.getUrl(filename);
+            if(cachedUrl && cachedUrl !== filename) {
```

Comment:
do you need the "if cachedUrl exists" check? Shouldn't the `cachedUrl !== filename` check suffice?